src/crypto/tls/common.go

```diff
@@ -338,6 +338,9 @@ type ConnectionState struct {
 	// accepted by the server.
 	ECHAccepted bool
 
+	// ECHOffered is set if the ECH extension was offered by the client.
```

Comment:
To match the verbage elsewhere in the code, this should say
```suggestion
	// ECHOffered is set if the ECH extension is present in the ClientHello.
	// This means the client has offered ECH or sent GREASE ECH.
```
Also, we might consider giving it a slightly less ambiguous name, like `ECHExtensionPresent` or something. Alternatively, you could add two flags, one for ECH offered and another for ECH greased.